argon2/src/lib.rs

```diff
@@ -62,13 +62,14 @@
 //! [key derivation function]: https://en.wikipedia.org/wiki/Key_derivation_function
 //! [Password Hashing Competition]: https://www.password-hashing.net/
 
-#![no_std]
+#![cfg_attr(not(feature = "parallel"), no_std)]
```

Comment:
Would suggest leaving this as-is:

```suggestion
#[no_std]
```

...and adding the following below the existing similar directive for `extern crate alloc`:

```rust
#[cfg(feature = "std")]
extern crate std;
```